src/main/java/edu/hm/hafner/analysis/parser/Gcc4CompilerParser.java

```diff
@@ -30,6 +30,14 @@ public Gcc4CompilerParser() {
         super(GCC_WARNING_PATTERN);
     }
 
+    /**
+     * Creates a new instance of {@link Gcc4CompilerParser} with specified pattern.
+     * @param pattern a regex pattern to be used instead of the default one
+     */
+    public Gcc4CompilerParser(final String pattern) {
+        super(pattern == null || pattern.isEmpty() ? GCC_WARNING_PATTERN : ANT_TASK + pattern);
```

Comment:
Fail fast if possible:
```suggestion
        super(pattern);
```

Reply:
should this be just `pattern` or `ANT_TASK + pattern`

Reply:
I think this should be solely `pattern`. I think that the deriving class should have full control over the pattern.

Reply:
OK, I'll add the ANT_TASK in the DoxygenParser class, same as it's Gcc4CompilerParser. Without it there were some tests that failed.